operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/util/CommonUtil.java

```diff
@@ -47,10 +48,16 @@ public void checkAllowed(FHIROperationContext operationContext, boolean isImport
             String source = adapter.getStorageProvider();
             String outcome = adapter.getStorageProviderOutcomes();
 
-            boolean s = ConfigurationFactory.getInstance().hasStorageProvider(source);
-            boolean o = ConfigurationFactory.getInstance().hasStorageProvider(outcome);
-            if (!s || !o) {
-                throw FHIROperationUtil.buildExceptionWithIssue("The storage provider for outcome [" + s + "] or source [" + o + "] passed is not configured properly" , IssueType.EXCEPTION);
+            if (!ConfigurationFactory.getInstance().hasStorageProvider(source)
+                    || !ConfigurationFactory.getInstance().hasStorageProvider(outcome)) {
+                String fhirTenant = FHIRRequestContext.get().getTenantId();
+                StringBuilder builder = new StringBuilder("The 'fhirServer/bulkdata/storageProvider' for source [");
+                builder.append(fhirTenant).append("/").append(source);
+                builder.append("] or outcome [");
+                builder.append(fhirTenant).append("/").append(outcome);
+                builder.append("] passed is not configured properly");
```

Comment:
can we say that they are missing or is there a chance is misconfigured in some other way?

Reply:
we can say it's missing. I opted not to do so as there is an opportunity to 'discover' valid providers... in an open environment it may lead to unintended consequences.

Reply:
i didn't follow this

Reply:
I've opted not to be specific in order to eliminate potential 'discovery' issues with the configuration